src/context/ModalProvider/types.ts

```diff
@@ -0,0 +1,57 @@
+import type { FC } from 'react'
+import type { BackupPassphraseModalProps } from 'components/Layout/Header/NavBar/Native/BackupPassphraseModal/BackupPassphraseModal'
+import type { AssetSearchModalProps } from 'components/Modals/AssetSearch/AssetSearchModal'
+import type { FiatRampsModalProps } from 'components/Modals/FiatRamps/FiatRampsModal'
+import type { NftModalProps } from 'components/Modals/Nfts/NftModal'
+import type { PopupWindowModalProps } from 'components/Modals/PopupWindowModal'
+import type { QrCodeModalProps } from 'components/Modals/QrCode/QrCode'
+import type { ReceivePropsType } from 'components/Modals/Receive/Receive'
+import type { SendModalProps } from 'components/Modals/Send/Send'
+
+import type { CLOSE_MODAL, OPEN_MODAL } from './constants'
+
+export type Modals = {
+  receive: ({ asset, accountId }: ReceivePropsType) => JSX.Element
+  qrCode: ({ assetId, accountId }: QrCodeModalProps) => JSX.Element
+  send: ({ assetId, accountId, input }: SendModalProps) => JSX.Element
+  fiatRamps: FC<FiatRampsModalProps>
+  settings: () => JSX.Element
+  keepKeyWipe: () => JSX.Element
+  backupNativePassphrase: FC<BackupPassphraseModalProps>
+  mobileWelcomeModal: () => JSX.Element
+  addAccount: () => JSX.Element | null
```

Comment:
Ditto for this one, `React.FC` infers this if we want to use it instead for consistency
<img width="556" alt="image" src="https://github.com/shapeshift/web/assets/17035424/dde89929-04da-4dcc-8c42-bca14a3a2884">